### PR TITLE
[TASK-7429] fix authentication 

### PR DESCRIPTION
**Pull request by jjramirezn**
## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for user validation and API responses.
	- Introduction of a new type for better type safety in API interactions.
	- Added functionality for cashout link submissions.

- **Bug Fixes**
	- Updated KYC status checks in multiple components to improve user experience.
	- Improved error messaging for unselected bank accounts and KYC status updates.

- **Documentation**
	- Updated documentation to reflect changes in method naming for clarity.

- **Refactor**
	- Renamed methods for better clarity and updated parameters accordingly.
	- Consolidated import statements for improved code organization.

- **Style**
	- Minor adjustments to CSS class orders for better styling consistency.

### CODE REVIEW

File: src/app/api/bridge/user/new/get-status/route.ts [hunk 1]
```diff
@@ -1,13 +1,34 @@
-import type { NextRequest } from 'next/server'
-import { NextResponse } from 'next/server'
+import { NextResponse, NextRequest } from 'next/server'
 import * as interfaces from '@/interfaces'
+import { GET as getUserFromCookie } from '@/app/api/peanut/user/get-user-from-cookie/route'
```
Comment (Hugo0) [inline src/app/api/bridge/user/new/get-status/route.ts:3]: eww

File: src/app/api/bridge/user/new/get-status/route.ts [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
 
 export async function POST(request: NextRequest) {
     try {
         const { userId, type } = await request.json()
         if (!process.env.BRIDGE_API_KEY) {
             throw new Error('BRIDGE_API_KEY is not defined')
         }
+
+        const getUserFromCookieRequest = new NextRequest(
```
Comment (Hugo0) [inline src/app/api/bridge/user/new/get-status/route.ts:12]: this plus ` { GET as getUserFromCookie }` ? oof

File: src/app/api/bridge/user/new/get-status/route.ts [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
+            `${process.env.NEXT_PUBLIC_BASE_URL}/api/peanut/user/get-user-from-cookie`,
+            {
+                method: 'GET',
+                headers: {
+                    cookie: request.headers.get('cookie') ?? '',
+                    ...request.headers,
+                },
+            }
+        )
+        const getUserFromCookieResponse = await getUserFromCookie(getUserFromCookieRequest)
```
Comment (Hugo0) [inline src/app/api/bridge/user/new/get-status/route.ts:22]: what's going on here, why

**Review by Hugo0 — APPROVED** [review]

File: src/app/api/bridge/user/new/get-status/route.ts (hunk 3 above)
Comment (jjramirezn) [reply src/app/api/bridge/user/new/get-status/route.ts:22]: We need the user data, of the calling user, so we need to ensure that the user is authenticated and get its data. We can't trust on input

**Pull request merged**